File: backend/app/schemas/listing.py
from datetime import datetime
from pydantic import BaseModel
from app.schemas.user import UserResponse
from typing import Literal

class ListingCreate(BaseModel):
    title: str
    description: str
    price: int
    mileage: int
    year: int
    location_city: str
    image_url: str
    status: Literal["active", "sold"]

class ListingResponse(BaseModel):
    id: int
    title: str
    description: str
    price: int
    mileage: int
    year: int
    location_city: str
    image_url: str
    status: Literal["active", "sold"]
    created_at: datetime
    user: UserResponse

    class Config:
        from_attributes = True

File: frontend/types/listing.ts
export type Listing = {
  id: number;
  title: string;
  price: number;
  location: string;
  image_url: string;
  description: string;
  mileage: number;
  year: number;
  sellerName: string;
  sellerPhone: string;
  status: "active" | "sold";
};
